sg_covid_impact/secondary_data.py

```diff
@@ -0,0 +1,105 @@
+_SIMD_URL = "https://www.gov.scot/binaries/content/documents/govscot/publications/statistics/2020/01/scottish-index-of-multiple-deprivation-2020-ranks-and-domain-ranks/documents/scottish-index-of-multiple-deprivation-2020-ranks-and-domain-ranks/scottish-index-of-multiple-deprivation-2020-ranks-and-domain-ranks/govscot%3Adocument/SIMD%2B2020v2%2B-%2Branks.xlsx"
+_DZ_LU = "http://statistics.gov.scot/downloads/file?id=2a2be2f0-bf5f-4e53-9726-7ef16fa893b7%2FDatazone2011lookup.csv"
+_SECOND = f"{project_dir}/data/processed/lad_secondary.csv"
+
+
+def fetch_process_nomis(
+    url, indicator_name, value_column, source, indicator_column="MEASURES_NAME"
+):
+    """Fetch nomis data
+    Args:
+        url (str): API url
+        indicator_name (str): name of indicator
+        value_column (str): value column
+        source (str): data source
+        indicator_column (str): column that contains the indicator
+    Returns:
+        A clean table with secondary data
+    """
+    logging.info(f"Fetching {source}")
+    df = pd.read_csv(url)
+    df_sel = df.query(f"{indicator_column}=='{indicator_name}'")[
+        ["DATE", "GEOGRAPHY_NAME", "GEOGRAPHY_CODE", value_column, "OBS_VALUE"]
+    ].reset_index(drop=True)
+    df_sel.rename(
+        columns={"OBS_VALUE": "VALUE", value_column: "VARIABLE"}, inplace=True
+    )
+    df_sel.columns = [x.lower() for x in df_sel.columns]
+
+    df_sel["source"] = source
+
+    return df_sel
+
+
+def process_simd(df, indices):
+    """Processes the Scottish Index of Multiple Deprivation data
+    Args:
+        df (df): A SIMD table
+        indices (list): indices we want to aggregate over
+    Returns:
+        Table with share of population living in high SIMD areas
+    """
+    logging.info("Fetching SIMD")
+    dz_lu = pd.read_csv(_DZ_LU)
+    datazone_lad_code_lookup = dz_lu.set_index("DZ2011_Code")["LA_Code"].to_dict()
+
+    # Calculate deciles of deprivation
+    df["decile"] = pd.qcut(
+        df["SIMD2020v2_Rank"], q=np.arange(0, 1.1, 0.1), labels=False, duplicates="drop"
+    )
+
+    df["geography_code"] = df["Data_Zone"].map(datazone_lad_code_lookup)
+
+    simd_agg = df.groupby(["Council_area", "decile", "geography_code"])[
+        "Total_population"
+    ].sum()
+
+    # Calculate shares of the population living in areas with different rankings
+    sim_pop_share = (
+        simd_agg.reset_index(name="population")
+        .pivot_table(
+            index=["Council_area", "geography_code"],
+            columns="decile",
+            values="population",
+        )
+        .apply(lambda x: x / x.sum(), axis=1)
+    )
+
+    sim_high_depr_share = sim_pop_share[[0, 1]].sum(axis=1).reset_index(name="value")
+
+    # Change column names
+    sim_high_depr_share["variable"] = "smd_high_deprivation_share"
+    sim_high_depr_share.rename(columns={"Council_area": "geography_name"}, inplace=True)
+    sim_high_depr_share["source"] = "simd"
+    sim_high_depr_share["date"] = 2020
+
```

Comment:
General observation: this function involves quite a few transformations and it's hard to keep track of the shape of the data throughout the function